### PR TITLE
Update to latest Matomo (formerly Piwik) list

### PR DESCRIPTION
**Pull request by dan-blanchard**
This PR changes all references to Piwik to say "Matomo" instead and also updates the search engine list to latest.

### CODE REVIEW

File: serpextract/search_engines.json [hunk 1]
```diff
@@ -1,94 +1,94 @@
...
-      "backlink":"s\/{k}",
-      "charsets":[
+      "backlink": "s/{k}",
+      "charsets": [
         "iso-8859-2"
       ],
-      "params":[
-        "\/s\\\/([^\\\/]+)\/",
+      "params": [
+        "/s\\/([^\\/]+)/",
```
Comment (aldraco) [inline serpextract/search_engines.json:9]: What does the removal of escape characters mean in this case? Is it an artifact of black reformatting or is it necessary to work correctly?

Comment (dan-blanchard) [reply serpextract/search_engines.json:9]: Those are changes made in the upstream list. I didn't change them. This is just a JSON file. 

Anyway, the escape characters they removed were unnecessary ones.

**Comment by emmett9001** [comment]
How did you create this JSON file? Matomo's list is YAML https://github.com/matomo-org/searchengine-and-social-list/blob/master/SearchEngines.yml

**Comment by cwisecarver** [comment]
It looks like `update_list.py` changes the format internally.

**Pull request merged**